src/main/scala/scorex/api/http/TransactionsApiRoute.scala

```diff
@@ -56,21 +57,122 @@ case class TransactionsApiRoute(
+                    complete(Json.arr(JsArray(state.accountTransactions(a, limit, 0).map { case (h, tx) =>
                       processedTxToExtendedJson(tx) + ("height" -> JsNumber(h))
+                    })))
+                  case Some(limit) if limit > MaxTransactionsPerRequest =>
+                    complete(TooBigArrayAllocation)
+                  case _ =>
+                    complete(invalidLimit)
+                }
+              }
+          }
+      }
+    }
+  }
+
+  private val invalidOffset = StatusCodes.BadRequest -> Json.obj("message" -> "invalid.offset")
+
+  @Path("/address/{address}/limit/{limit}/offset/{offset}")
+  @ApiOperation(value = "Address", notes = "Get list of transactions where specified address has been involved", httpMethod = "GET")
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "address", value = "Wallet address ", required = true, dataType = "string", paramType = "path"),
+    new ApiImplicitParam(name = "limit", value = "Specified number of records to be returned", required = true, dataType = "integer", paramType = "path"),
+    new ApiImplicitParam(name = "offset", value = "Specified number of records offset", required = true, dataType = "integer", paramType = "path")
+  ))
+  def addressLimitOffset: Route = (pathPrefix("address") & get) {
+    pathPrefix(Segment) { address =>
+      Address.fromString(address) match {
+        case Left(e) => complete(ApiError.fromValidationError(e))
+        case Right(a) =>
+          pathPrefix("limit") {
+            pathEndOrSingleSlash {
+              complete(invalidLimit)
+            } ~
+              pathPrefix(Segment) { limitStr =>
+                Exception.allCatch.opt(limitStr.toInt) match {
+                  case Some(limit) if limit > 0 && limit <= MaxTransactionsPerRequest =>
+                    pathPrefix("offset") {
+                      pathEndOrSingleSlash {
+                        complete(invalidOffset)
+                      } ~
+                        path(Segment) { offsetStr =>
+                          Exception.allCatch.opt(offsetStr.toInt) match {
+                            case Some(offset) if offset >= 0 && offset <= MaxTransactionOffset =>
+                              complete(Json.arr(JsArray(state.accountTransactions(a, limit, offset).map { case (h, tx) =>
+                                processedTxToExtendedJson(tx) + ("height" -> JsNumber(h))
+                              })))
+                            case  _ =>
+                              complete(invalidOffset)
+                          }
+                        }
                     }
-                    )))
                   case Some(limit) if limit > MaxTransactionsPerRequest =>
                     complete(TooBigArrayAllocation)
                   case _ =>
                     complete(invalidLimit)
                 }
               }
-          } ~ complete(StatusCodes.NotFound)
+          }
```

Comment:
what's different without add `complete(StatusCodes.NotFound)`?

Reply:
L69 will block the longer route than '/limit/{limit}'.